### PR TITLE
Share buttons for social networks

### PR DESCRIPTION
**Pull request by mxmehl**
First attempt to implement share buttons for social networks. On the index page, they are present on the left side. On event sites, they are visible as buttons under the data. I chose Fediverse (Mastodon, Diaspora...), Twitter, Reddit and Facebook.

Caveats so far:
* I cannot figure out how to calculate the URL for an event page from within the template. This should be present in L163 in event.html
* Having both horizontal and vertical share buttons causes some stupid errors with the CSS: Should be easy to figure out, but unless we device which buttons to use where, I won't invest more time into it :)

### CODE REVIEW

**Comment by dreirik** [comment]
thanks @mxmehl for the  share buttons. How is it possible I can see them? Would it be possible to put them on the test-instance?

Also I wonder where can I see the text that is generated by the share buttons?

**Comment by weeman1337** [comment]
I found time to test it. Here is my feedback:

- I would do the share icons on the home page like on the detail pages. It's easier especially when it comes to responsive design.
- Is it possible to add an e-mail share button?
- We now promote proprietary services like Twitter and Facebook. We've talked about that it's okay. But in my opinion we then should also promote some free services like "Mastodon". Yes it's already behind the "Fediverse" button - but kind of hidden. Long talk: Could we add a dedicated "Mastodon" button?

File: src/cgi-bin/share.php [hunk 1]
```diff
@@ -0,0 +1,147 @@
+<?php
```
Comment (weeman1337) [inline src/cgi-bin/share.php]: Can we move that to `src/php`? The build script could just copy it to the root..?

**Comment by weeman1337** [comment]
For some nice hover animation and better mouse cursors we should update the share buttons once https://git.fsfe.org/FSFE/share-buttons/pulls/2 is merged.

File: src/styles/buttons-side.css [hunk 2]
```diff
@@ -0,0 +1,136 @@
+/*
```
Comment (weeman1337) [inline src/styles/buttons-side.css]: Have a look at the PR in the upstream repo

**Comment by mxmehl** [comment]
> thanks @mxmehl for the share buttons. How is it possible I can see them? Would it be possible to put them on the test-instance?

@eal, please read the README to learn how to preview the website on your computer. regarding the test website, this PR asks to actually include it there, but local preview is the best way IMHO

> Also I wonder where can I see the text that is generated by the share buttons?

you mean on the actual services? You need to have php installed on your computer. Then you could navigate in your terminal to the `build` directory (after you have built the site locally), and run `php -S localhost:8000`. Then, in your browser, open http://localhost:8000/index.html, where you then can click on the buttons.

Please note:
* This will only work after my next PR as I will only then remove the dependency on Apache redirections, as rightfully suggested by Michael
* This is an alternative approach to preview the website in your browser. The README instructions open a local webserver on port 1337, but that does not support PHP afaik

**Comment by mxmehl** [comment]
Thanks for your feeback, @weeman1337!

>     * I would do the share icons on the home page like on the detail pages. It's easier especially when it comes to responsive design.

OK, do you have a suggestion where to put it?

>     * Is it possible to add an e-mail share button?

Oh yes, I've added that upstream.

>     * We now promote proprietary services like Twitter and Facebook. We've talked about that it's okay. But in my opinion we then should also promote some free services like "Mastodon". Yes it's already behind the "Fediverse" button - but kind of hidden. Long talk: Could we add a dedicated "Mastodon" button?

I tend to disagree. If we add Mastodon, people will complain why we don't mention GNUsocial, Friendica or Diaspora directly. And it requires people to know which services they signed up with. Fediverse is neutral, and allows us to add new fediverse instances in share.php without having to add their buttons.

>     * Maybe we should pack all the css into one file.

Absolute. Please see upstream for the changes.

>     * Can we just keep the share thing `share.php` or is there any important reason rewriting `share` to `share.php`? Then we can work with a local PHP dev server. Now it requires a local Apache server.

Just for links beauty reasons. but your argument convinces me, I will remove that as soon as you feel that upstream looks fine enough

File: src/cgi-bin/share.php (hunk 1 above)
Comment (mxmehl) [reply src/cgi-bin/share.php]: Sure. What do you think about adding the whole thing as a submodule?

File: src/styles/buttons-side.css (hunk 2 above)
Comment (mxmehl) [reply src/styles/buttons-side.css]: Done

**Comment by weeman1337** [comment]
> I tend to disagree. If we add Mastodon, people will complain why we don't mention GNUsocial, Friendica or Diaspora directly. And it requires people to know which services they signed up with. Fediverse is neutral, and allows us to add new fediverse instances in share.php without having to add their buttons.

I tend to disagree :D Ask people whether they know Mastodon or the Fediverse. Guess the results ;) Anyway let's discuss these points later.

> Just for links beauty reasons. but your argument convinces me, I will remove that as soon as you feel that upstream looks fine enough

Upstream looks fine enough.

**Comment by weeman1337** [comment]
> OK, do you have a suggestion where to put it?

Maybe below the header. Let's do that later.

See also https://github.com/foss-events/website/issues/42

**Comment by dreirik** [comment]
> > thanks @mxmehl for the share buttons. How is it possible I can see them? Would it be possible to put them on the test-instance?
> 
> @eal, please read the README to learn how to preview the website on your computer. regarding the test website, this PR asks to actually include it there, but local preview is the best way IMHO

ok, I followed the readme but still I cannot see the share buttons on my localhost. I tried to go into the test-branch and build  local preview there but this did not work out at all, saying "make: *** No rule to make target 'src/lib/share-buttons/buttons.css', needed by 'build/styles/buttons.css'.  Stop."

**Comment by mxmehl** [comment]
please check out the test branch (which holds the latest changes) and run `make clean; make`. the clean is necessary to get rid of old files first.

**Pull request merged**